### PR TITLE
Fix adsc wrong syncCh size

### PR DESCRIPTION
**Pull request by YonkaFang**
**Please provide a description of this PR:**

This PR fix a bug in adsc. Which will cause adsc hang when having received initial resource push.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
@YonkaFangIt looks `syncCh` is read from `WaitConfigSync` and not used in istio, do you use it in your own code base?

**Comment by YonkaFang** [comment]
> @YonkaFangIt looks `syncCh` is read from `WaitConfigSync` and not used in istio, do you use it in your own code base?

Currently adsc is used in mcp-over-xds configcontroller is istio(rather than our own code), but the func  `WaitConfigSync` is not called. I think there's some work for configController to do.

If syncCh is set to a correct size, adsc can normally works even when nobody cares abound the sync status/read the syncCh.

File: releasenotes/notes/34633.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+
+releaseNotes:
+- |
+  **Fixed** adsc wrong syncCh size which will make it be certain to hang
```
Comment (brian-avery) [inline releasenotes/notes/34633.yaml:7]:
```suggestion
  **Fixed** an issue in which ADS would hang due to the wrong `syncCh` size being provided.
```

**Review by brian-avery — APPROVED** [review]

**Comment by brian-avery** [comment]
Looks good to me. Please take a look at the release note suggestion before removing the `do-not-merge/hold` label.

**Comment by YonkaFang** [comment]
@brian-avery plz take a look. suggested release note change has been applied.

**Comment by hzxuzhonghu** [comment]
@YonkaFang I come to think maybe it's better to remove the `syncCh` and the `WaitConfigSync `

**Comment by hzxuzhonghu** [comment]
With just a HasSync method, it allows users make use of it easily like other controllers and config store do

**Comment by YonkaFang** [comment]
> and

I also thinks the user-side should do the wait thing and keep the consistency among the controllers. I'm ajust afraid that anyone will use this func later.

**Review by hzxuzhonghu — APPROVED** [review]
This is not a api, we didnot promise our package does not change. Actually, we are always changing internally.

**Pull request merged**